app/controllers/telegram_webhooks_controller.rb

```diff
@@ -10,73 +10,42 @@ def start(*)
   end
 
   def shows(*args)
-    endingDate = Date.parse(END_DATE_STRING, t(:date_format))
-
     # If arguments is not empty, check if it's a number or a string
     if args.any?
-      #Day informed as a number
-      if args[0].match(NUMBER_REGULAR_EXPRESSION)
-        dayFromArgs = args[0].to_s.rjust(2, '0')
-        errorMessage = validateDayFromMonth(dayFromArgs.to_i)
-      elsif args[0].is_a? String
-        #Day informed as a string (monday, tuesday...)
-        dayFromArgs = getDayFromString(args[0].to_s)
-      else
-        #Command does not match anything
-        errorMessage = MESSAGE_CHECK_USAGE_COMMAND
-      end
+
+      selectedDay, argsTypeErrorMessage, selectedMonth = getArgsFromCommand(*args)
 
       # Return immediately if an error message was caught
-      if errorMessage
-        reply_with :message, text: errorMessage
+      if argsTypeErrorMessage
+        replyWithText(argsTypeErrorMessage)
         return
-      end
-
-      if not dayFromArgs
-        reply_with :message, text: t(:error_invalid_weekday)
+      elsif selectedDay.nil?
```

Comment:
How bad is to use the current solution? Idk if it's just weird or something else...